### PR TITLE
Wrap more Epetra_Comm calls

### PR DESCRIPTION
**Pull request by sebproell**
Part of #82 

This PR replaces most functions of the `Epetra_MpiComm` with our own wrapper version and drops the use of `Epetra_SerialComm`. This change uncovered an existing memory bug in Xfluid which I also fixed here.

### CODE REVIEW

File: src/xfem/4C_xfem_xfluid_contact_communicator.cpp [hunk 1]
```diff
@@ -1478,7 +1478,7 @@ void XFEM::XFluidContactComm::create_new_gmsh_files()
 #endif
 
   std::vector<int> g_sum_gps(5);
-  fluiddis_->get_comm().SumAll(sum_gps_.data(), g_sum_gps.data(), 5);
+  Core::Communication::sum_all(sum_gps_.data(), g_sum_gps.data(), 5, fluiddis_->get_comm());
```
Comment (sebproell) [inline src/xfem/4C_xfem_xfluid_contact_communicator.cpp:1481]: There is no resize operation on `sum_gps_`, so its size is not going to be 5. This invokes UB.

Comment (sebproell) [reply src/xfem/4C_xfem_xfluid_contact_communicator.cpp:1482]: I now resized this vector appropriately. Maybe this fixes one of the famous random xfluid failures.

**Review by amgebauer — APPROVED** [review]

**Review by c-p-schmidt — APPROVED** [review]

**Review by mayrmt — APPROVED** [review]
@sebproell Looks good! 

Will there be a point in the near future, where one must never store a pointer to the communicator or its properties such as size or rank on any class? I believe we do have quite some `comm_`, `my_rank_`, ... class members.

**Comment by sebproell** [comment]
@mayrmt 

> Will there be a point in the near future, where one must never store a pointer to the communicator or its properties such as size or rank on any class? I believe we do have quite some `comm_`, `my_rank_`, ... class members. 

The only change I mean to make with #82, is ro replace all Epetra_Comm occurrences with MPI_Comm. You can and should still store or pass the relevant communicator where needed. It's just a different and simpler type which is not part of Epetra. While we do this, we might as well change the interface of our MPI functions to more modern C++, but this is a side project for the Tpetra migration.

**Pull request merged**